src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1000 @@
+    !> \details
+    !> Generate the name of the file that contains the coordinate lat, lon 
+    !> (in radians) and optionally return start_x and start_y (the location 
+    !> of the first global pixel coordinate of tile). Warning: This function
+    !> can produce filenames that may not exist (For lon less than -.5 * pi and 
+    !> greater than .5 * pi and lat less than -pi and greater than pi).
+    !
+    !-----------------------------------------------------------------------
+    function mpas_geotile_mgr_gen_tile_name(mgr, lat, lon, fname, start_x, start_y) result(ierr)
+
+        implicit none
+
+        class (mpas_geotile_mgr_type) :: mgr
+        real (kind=RKIND), value :: lat
+        real (kind=RKIND), value :: lon
+        character (len=StrKIND), intent(out) :: fname
+        integer, intent(out), optional :: start_x
+        integer, intent(out), optional :: start_y
+
+        character (len=StrKIND), parameter :: fname_format = "(i5.5, '-', i5.5, '.', i5.5, '-', i5.5)"
+
+        real (kind=RKIND), pointer :: dx
+        real (kind=RKIND), pointer :: dy
+        integer, pointer :: tile_nx
+        integer, pointer :: tile_ny
+        integer, dimension(2) :: x
+        integer, dimension(2) :: y
+
+        integer :: ierr
+        ierr = 0
+
+        call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+        call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        ! Find the global pixel location that contains lat, lon
+        call mgr % latlon_to_pixel(lat, lon, x(1), y(1))
+
+        ! Calculate the range of this tile, which will be its filename
+        x(1) = (x(1) - modulo(x(1), tile_nx)) + 1
+        x(2) = x(1) + tile_nx
+        x(2) = x(2) - modulo(x(2), tile_nx)
+
+        y(1) = (y(1) - modulo(y(1), tile_ny)) + 1
+        y(2) = y(1) + tile_ny
+        y(2) = y(2) - modulo(y(2), tile_ny)
+
+        write(fname, fname_format) x(1), x(2), y(1), y(2)
+
+        if (present(start_x)) then
+            start_x = x(1)
+        endif
+        if (present(start_y)) then
+            start_y = y(1)
+        endif
+
+    end function mpas_geotile_mgr_gen_tile_name
+
+
+    !***********************************************************************
+    !
+    !  public subroutine mpas_geotile_mgr_tile_to_latlon => tile_to_latlon
+    !
+    !> \brief   Translate a tile indices to latitude and longitude,
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Given a tile, translate the relative tile coordinates, i, j, of that
+    !> tile to a latitude and longitude coordinate. Upon success, lat, lon
+    !> will be in the range of -1/2 * pi to 1/2 * pi and 0 to 2.0 * pi, respectively.
+    !
+    !-----------------------------------------------------------------------
+    subroutine mpas_geotile_mgr_tile_to_latlon(mgr, tile, j, i, lat, lon)
```

Comment:
When testing this function, I was a bit confused by the range of latitudes that it produced. If I grabbed the most north east tile `mgr % get_tile(5. * pi, 2.0 * pi, tile)` some of the range of latitudes produced by `tile_to_latlon` were greater than .5 * pi, and I'm not sure exactly why that would be.

A first thought is that this is because `known_lat` is not equal to exactly 5. * -pi. @mgduda, is that correct?

Reply:
Typically, the `known_lat` and `known_lon` values for a dataset give the coordinates of the *center* of a pixel. So if the boundary of the pixel extends to, e.g., the South Pole, then the pixel's center coordinates will be greater than -Pi/2.

Reply:
I just left a comment for lines 642 and 643 that may be related.